torch/export/_trace.py

```diff
@@ -1531,13 +1536,18 @@ def wrapped_fn(*args):
 
     flat_args = pytree.tree_leaves((fake_args, fake_kwargs))
     index = 0
+    fake_mode = detect_fake_mode(flat_args)
```

Comment:
This code looks the same as one before. Could we make it a util function?

Reply:
A lot of the code here is duplicate. Do you want me to pull out more or just this code?

Reply:
I think more the better :) But for this PR, just this part is fine.